dvc/fs/oss.py

```diff
@@ -8,124 +8,73 @@
-    """
-
+# pylint:disable=abstract-method
+class OSSFileSystem(ObjectFSWrapper):
     scheme = Schemes.OSS
     PATH_CLS = CloudURLInfo
-    REQUIRES = {"oss2": "oss2"}
+    REQUIRES = {"ossfs": "ossfs"}
     PARAM_CHECKSUM = "etag"
     COPY_POLL_SECONDS = 5
     LIST_OBJECT_PAGE_SIZE = 100
+    DETAIL_FIELDS = frozenset(("etag", "size"))
 
-    def __init__(self, **config):
-        super().__init__(**config)
-
-        self.endpoint = config.get("oss_endpoint") or os.getenv("OSS_ENDPOINT")
-
-        self.key_id = (
-            config.get("oss_key_id")
-            or os.getenv("OSS_ACCESS_KEY_ID")
-            or "defaultId"
+    def _prepare_credentials(self, **config):
+        login_info = {}
+        login_info["key"] = config.get("oss_key_id") or os.getenv(
+            "OSS_ACCESS_KEY_ID"
         )
-
-        self.key_secret = (
-            config.get("oss_key_secret")
-            or os.getenv("OSS_ACCESS_KEY_SECRET")
-            or "defaultSecret"
+        login_info["secret"] = config.get("oss_key_secret") or os.getenv(
+            "OSS_ACCESS_KEY_SECRET"
         )
+        login_info["endpoint"] = config.get("oss_endpoint")
+        return login_info
 
     @wrap_prop(threading.Lock())
     @cached_property
-    def oss_service(self):
-        import oss2
-
-        logger.debug(f"key id: {self.key_id}")
-        logger.debug(f"key secret: {self.key_secret}")
-
-        return oss2.Auth(self.key_id, self.key_secret)
-
-    def _get_bucket(self, bucket):
-        import oss2
-
-        return oss2.Bucket(self.oss_service, self.endpoint, bucket)
+    def fs(self):
+        from ossfs import OSSFileSystem as _OSSFileSystem
 
-    def _generate_download_url(self, path_info, expires=3600):
-        return self._get_bucket(path_info.bucket).sign_url(
-            "GET", path_info.path, expires
-        )
-
-    def exists(self, path_info) -> bool:
-        paths = self._list_paths(path_info)
-        return any(path_info.path == path for path in paths)
-
-    def _list_paths(self, path_info):
-        import oss2
-
-        for blob in oss2.ObjectIterator(
-            self._get_bucket(path_info.bucket), prefix=path_info.path
-        ):
-            yield blob.key
-
-    def walk_files(self, path_info, **kwargs):
-        if not kwargs.pop("prefix", False):
-            path_info = path_info / ""
-        for fname in self._list_paths(path_info):
-            if fname.endswith("/"):
-                continue
-
-            yield path_info.replace(path=fname)
+        return _OSSFileSystem(**self.fs_args)
 
     def remove(self, path_info):
-        if path_info.scheme != self.scheme:
-            raise NotImplementedError
-
-        logger.debug(f"Removing oss://{path_info}")
-        self._get_bucket(path_info.bucket).delete_object(path_info.path)
-
-    def _upload_fobj(self, fobj, to_info, **kwargs):
-        self._get_bucket(to_info.bucket).put_object(to_info.path, fobj)
+        self.fs.rm_file(self._with_bucket(path_info))
```

Comment:
Why do you re-implement this? 
https://github.com/iterative/dvc/blob/f45311e658f4ce0ec537dfc421a4c45aa8efb13e/dvc/fs/fsspec_wrapper.py#L112-L113

Reply:
`rm` equals to `expand` and `rm_file` which is slower than directly `rm_file` if we had already known the removed path_info is a `object`.

Reply:
Right. For a single file the behavior should be close, but it is possible that `expand_paths` calls on some providers might be a little bit problematic. Apparently s3fs doesn't implement `_rm_file()`, so it is a bit problematic to just switch it of on the `FsspecWrapper` but once it is done we should be safe with replacing this for all providers. Nice catch!

Reply:
Yes looks like s3fs has no rm_file.

https://github.com/intake/filesystem_spec/blob/6de675e24adcbc1a8c3d74f5956a79bc95c34c2b/fsspec/spec.py#L884

https://github.com/dask/s3fs/blob/fc4489cfe749b2d3de62bde926ed93c05e27ce7a/s3fs/core.py#L1640

But I think that the best solution for `gc` is to call a `batch_delete` ( don't know the exact name of it in other remotes).

Reply:
Yea, that is among my plans. But we first probably want to go with parallelization of gc and then add this special layer which is going to be tricky since only a handful of providers support it.